### PR TITLE
feat: Add react-native authentication

### PR DESCRIPTION
**Pull request by doubleface**
Add authentication method for react native environment

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -2429,6 +3005,162 @@
   resolved "https://registry.yarnpkg.com/@popperjs/core/-/core-2.9.1.tgz#7f554e7368c9ab679a11f4a042ca17149d70cf12"
   integrity sha512-DvJbbn3dUgMxDnJLH+RZQPnXak1h4ZVYQ7CWiFWjQwBFkVajT4rfw2PdpHLTSTwxrYfnoEXkuBiwkDm6tPMQeA==
 
+"@react-native-community/cli-debugger-ui@^5.0.1-alpha.1":
```
Comment (Crash--) [inline yarn.lock:3008]: peerDeps are changing the lock file? 🤔

File: packages/cozy-client/src/CozyClient.js [hunk 2]
```diff
@@ -1236,7 +1239,14 @@ client.query(Q('io.cozy.bills'))`)
...
     const stackClient = this.getStackClient()
     stackClient.setUri(cozyURL)
-    return this.startOAuthFlow(authenticateWithCordova)
+    const authFunction = this.isReactNative
+      ? authenticateWithReactNativeInAppBrowser
+      : authenticateWithCordova
+    return this.startOAuthFlow(authFunction)
+  }
+
+  isReactNative() {
```
Comment (Crash--) [inline packages/cozy-client/src/CozyClient.js:1248]: I think we should move the method to `src/authentication/mobile.js` 

Also, I think it more readable and extendable to have a `getAuthenticationMethod` in authent/mobile.js , and use it in `register` and `renew`. 

Like that, if we have to add others methods, it'll be easier. 

(also, `isReactNative` should not be part of the client API)

File: packages/cozy-client/src/authentication/mobile.js [hunk 3]
```diff
@@ -80,6 +82,51 @@ const authenticateWithInAppBrowser = url => {
   })
 }
 
+export const authenticateWithReactNativeInAppBrowser = url => {
```
Comment (Crash--) [inline packages/cozy-client/src/authentication/mobile.js:85]: js doc

File: packages/cozy-client/src/CozyClient.js [hunk 4]
```diff
@@ -1283,7 +1293,10 @@ client.query(Q('io.cozy.bills'))`)
    * @returns {object}   Contains the fetched token and the client information.
    */
   renewAuthorization() {
-    return this.authorize(authenticateWithCordova)
+    const authFunction = this.isReactNative
+      ? authenticateWithReactNativeInAppBrowser
+      : authenticateWithCordova
+    return this.authorize(authFunction)
```
Comment (Crash--) [inline packages/cozy-client/src/CozyClient.js:1299]: tests should be added for renew and register ;)

**Review by Crash-- — APPROVED** [review]

File: packages/cozy-client/src/authentication/mobile.js [hunk 5]
```diff
@@ -84,6 +99,15 @@ const authenticateWithInAppBrowser = url => {
   })
 }
 
+/**
+ * Open a RactNative InAppBrowsr
```
Comment (ptbrowne) [inline packages/cozy-client/src/authentication/mobile.js:103]: There are some typos:

Opens a ReactNative InAppBrowser and resolves with the URL
containing the token

File: packages/cozy-client/src/authentication/mobile.js [hunk 6]
```diff
@@ -50,7 +58,14 @@ const authenticateWithSafari = url => {
     }
   })
 }
-
+/**
+ * Open an inAppBrowsr
+ * and resolve with the URL containing
+ * the token
```
Comment (ptbrowne) [inline packages/cozy-client/src/authentication/mobile.js:64]: Opens
InAppBrowser
Resolves

I find it a bit less
readable when
sentence are split very often
by line breaks ;)

**Pull request merged**